documentation/correctness.adoc

```diff
@@ -0,0 +1,97 @@
+====
+
+It makes sense to phrase the answer as a collection of _issues_, instead of giving a hard "`yes/no`" answer.
+An issue is a specific point in the model where a specific kind of incorrectness arises.
+Many (kinds of) issues can be meaningfully recovered from, and recovery can often be obvious to the point it can be automatic.
+The unresolved reference can be left in as-is, and treated as an unknown (but not a _missing_) value with special meaning (i.e., domain-specific semantics).
+The illegal comma should be reported but can subsequently be safely ignored.
+In cases where recovery is not automatic, an issue should be very clear to the user what they can do to resolve it.
+A model must be editable/mutable -- and even usable to some -- even if it has issues.
+
+LionWeb provides a precise notion for model correctness in the form of a multi-leveled specification.
+For each level, it specifies what kinds of incorrectness can arise, how these should be reported as issues, and whether and how these can be meaningfully _recovered_ from.
+This approach provides a degree of resilience: issues arising in one level don't necessarily prohibit determining whether a model exhibits incorrectness on another level, provided an issue can be meaningfully _recovered_ from.
+
+LionWeb provides a JSON serialization format for models: the serialization chunk.
+LionWeb intentionally doesn't prescribe anything about the _programmatic representation_ of models -- i.e.: the runtime or in-memory representation that's the result of deserializing a serialization chunk of a model -- leaving implementors of language-oriented tooling free to choose the representation that's right for them.
+Because of that, this specification is phrased _completely_ in terms of the JSON serialization.
+The first three levels of this specification pertain purely to the JSON that's supposed to be the serialization chunk of a model, and are language(s)-aspecific.
+The next levels _could_ be phrased in terms of a programmatic representation, but the equivalent formulation purely in terms of the serialization format works just as well.
+
+This specification admits some flexibility on the first three, JSON-centric levels.
+That is because the behavior and adaptability of JSON parsers differ from implementation to implementation.
+We simply can't (and don't want to demand) that a LionWeb implementation always produces issues, even if they're recoverable or ignorable anyway.
+
+
+== Overview of the levels
+
+We first give an overview of the various levels, and what each pertains to, at first without specifying the corresponding kinds of issues that may arise in that level.
+The list is roughly in order from low-level to language(s)-specific.
+
+[horizontal]
+
+Low-level syntactic::
+Pertains to the character stream that's supposed to be JSON text that's (supposed to be) the serialization chunk of a model.
```

Comment:
Can we simplify this text? This is a summary, so readability is more important than absolute accuracy.